File: src/main/java/ma/youcode/myrh/models/JobOffer.java
package ma.youcode.myrh.models;


import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;
import java.util.List;

@Data
@Entity
@Table(name = "joboffers")
public class JobOffer{
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String description;
    private String city;
    private String profile;
    private String educationalLevel;
    private String salary;

    @Enumerated(EnumType.STRING)
    private Status status;

    @ManyToOne
    @JoinColumn(name = "recruiter_id")
    private Recruiter recruiter;
}

File: src/app/model/JobOffer.ts
import { Status } from "./JobOffer copy";

export interface JobOffer {
    id?: number | null;
    title: string | null;
    description: string | null;
    city: string | null;
    profile: string | null;
    educationalLevel: string | null;
    salary: string;
    status: string;
  }
  